src/controllers/UsersController.php

```diff
@@ -167,9 +173,22 @@ public function actionLogin(): ?Response
         $user = Craft::$app->getUsers()->getUserByUsernameOrEmail($loginName);
 
         if (!$user || $user->password === null) {
-            // Delay again to match $user->authenticate()'s delay
-            Craft::$app->getSecurity()->validatePassword('p@ss1w0rd', '$2y$13$nj9aiBeb7RfEfYP3Cum6Revyu14QelGGxwcnFUKXIrQUitSodEPRi');
-            return $this->_handleLoginFailure(User::AUTH_INVALID_CREDENTIALS);
+            
+            $event = new LoginUserNotFoundEvent([
+                'loginName' => $loginName,
+            ]);
+            $this->trigger(self::EVENT_LOGIN_USER_NOT_FOUND, $event);
```

Comment:
I don’t think this should be triggered in the event that `$user` was found but `$user->password === null`. Maybe just check for `if (!$user)` here, and then add a separate `if ($user->password === null)` check after it, keeping the original conditional block inside it.

Reply:
Yeah I understand.. Chang on it's way